Modules/timemodule.c

```diff
@@ -2390,3 +2398,44 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+#ifndef MS_WINDOWS
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+static int
+pysleep_zero_posix(void)
+{
+    static struct timeval zero = {0, 0};
+
+    PyTime_t deadline, monotonic;
+    if (PyTime_Monotonic(&monotonic) < 0) {
+        return -1;
+    }
+    deadline = monotonic;
+    do {
+        int ret, err;
+        Py_BEGIN_ALLOW_THREADS
+        ret = select(0, (fd_set *)0, (fd_set *)0, (fd_set *)0, &zero);
+        err = errno;
+        Py_END_ALLOW_THREADS
+        if (ret == 0) {
+            break;
+        }
+        if (err != EINTR) {
+            errno = err;
```

Comment:
It might be worth asserting that the function is called without an errno set.

Reply:
Considering it's at the very bottom of the file and that it's really an implementation detail, I wouldn't bother about it. I'm not sure the existing code checked it. But I'll add an assert at the beginning.

Reply:
See https://github.com/python/cpython/pull/128274#issuecomment-2563593057.